pinot-core/src/main/java/com/linkedin/pinot/core/segment/index/loader/invertedindex/InvertedIndexHandler.java

```diff
@@ -91,7 +91,31 @@ public void createInvertedIndices()
 
   private void createInvertedIndexForColumn(ColumnMetadata columnMetadata)
       throws IOException {
+    // Validation check
     String column = columnMetadata.getColumnName();
+    int totalDocs = columnMetadata.getTotalDocs();
+    if (totalDocs > OffHeapBitmapInvertedIndexCreator.MAX_NUM_ENTRIES) {
+      LOGGER.warn(
+          "Skip creating inverted index for segment: {}, column: {} because totalDocs: {} exceeds the limit: {}",
+          segmentName, column, totalDocs, OffHeapBitmapInvertedIndexCreator.MAX_NUM_ENTRIES);
+      return;
+    }
+    int cardinality = columnMetadata.getCardinality();
+    if (cardinality > OffHeapBitmapInvertedIndexCreator.MAX_NUM_ENTRIES) {
```

Comment:
Duplicated code logic. Move this to OffheapBitmapInvertedIndexCreator.isConstructionPossible(cardinality, totalDocs)  static method. That will keep all the logic in one place